Dialog/MainDialog.xaml.cs

```diff
@@ -184,18 +200,24 @@ private void CheckUnsafeFiles()
             HashSet<string> notsafe = GetHashSet(_config.UnsafeFiles);
 
             foreach (Outlook.Attachment item in _mail.Attachments)
-            {
-                foreach (string keyword in notsafe)
+            {
+                HashSet<string> seen = new HashSet<string>();
+                try
                 {
-                    if (item.FileName.Contains(keyword))
-                    {
+                    foreach (Match match in Regex.Matches(item.FileName, _config.UnsafeFilesPattern, RegexOptions.IgnoreCase))
+                    {
+                        if (seen.Contains(match.Value))

```

Comment:
```suggestion
                        string lowerValue = match.Value.ToLower();
                        if (seen.Contains(lowerValue))
```

`Regex.Matches`で`RegexOptions.IgnoreCase`を指定しているので、`item.FileName`及びそのマッチ結果であるmatch.Valueは大文字小文字どちらかに統一されておらず、大文字小文字違いがあった場合に別の値と判定されてしまいそうに見えます。

Reply:
ご指摘ありがとうございます。見落としておりました。